### PR TITLE
Add custom exceptions to add_or_get_filter

### LINKED ISSUE #7: Add custom exceptions for every error in add_or_get_filter
Instead of throwing a `ValueError`, throw a custom exception depending on the error code

### PR DESCRIPTION
**Pull request by domerd**
Closes issue #7.
Ran Black.

### CODE REVIEW

File: marine/marine.py [hunk 1]
```diff
@@ -81,9 +86,13 @@ def filter_and_parse(
                 bpf, display_filter, encoded_fields, encapsulation_type
             )
             if filter_id < 0:
-                raise ValueError(
-                    err
-                )  # TODO: create custom exception for every error type
+                if filter_id == -1:
```
Comment (tomlegkov) [inline marine/marine.py]: I think we should continue the trend of importing consts from the `.so` file, and also import these consts.

File: marine/marine.py [hunk 2]
```diff
@@ -4,6 +4,11 @@
 from io import StringIO
 from typing import Optional, List, Dict
 
+from marine.exceptions import (
```
Comment (tomlegkov) [inline marine/marine.py]: Relative import

**Review by tomlegkov — CHANGES_REQUESTED** [review]

File: marine/marine.py [hunk 3]
```diff
@@ -81,9 +86,13 @@ def filter_and_parse(
...
-                raise ValueError(
-                    err
-                )  # TODO: create custom exception for every error type
+                if filter_id == -1:
+                    raise BadBPFException(err)
+                elif filter_id == -2:
+                    raise BadDisplayFilterException(err)
+                elif filter_id == -3:
+                    raise InvalidFieldException(err)
+                raise ValueError(err)
```
Comment (yehonatanz) [inline marine/marine.py]: Shouldn't we use a custom exception here as well?

**Review by yehonatanz — CHANGES_REQUESTED** [review]

File: marine/marine.py [hunk 4]
```diff
@@ -81,9 +87,19 @@ def filter_and_parse(
                 bpf, display_filter, encoded_fields, encapsulation_type
             )
             if filter_id < 0:
-                raise ValueError(
-                    err
-                )  # TODO: create custom exception for every error type
+                if filter_id == c_int.in_dll(self._marine, "BadBPFErrorCode").value:
+                    raise BadBPFException(err)
```
Comment (yehonatanz) [inline marine/marine.py:91]: Looks great, juts change the casing here after you change it in the C code

**Review by yehonatanz — CHANGES_REQUESTED** [review]

**Review by yehonatanz — APPROVED** [review]

**Review by tomlegkov — APPROVED** [review]

**Review by shaharyarn — APPROVED** [review]

**Pull request merged**